Prefer std::ceil, std::floor, and std::round over their C counterparts

LumioseSil opened the pull request:
#### bb990589d735c8e0c6bd0049c1f6f8f78c7f4fd2
<pre>
Prefer std::ceil, std::floor, and std::round over their C counterparts
<a href="https://bugs.webkit.org/show_bug.cgi?id=253963">https://bugs.webkit.org/show_bug.cgi?id=253963</a>

Reviewed by NOBODY (OOPS!).

Unlike the C library versions, these std functions have overloads for
every type, so one does not need to worry about which version of the
function to call.

Continuation of https://commits.webkit.org/261584@main

* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
(JSC::B3::ConstFloatValue::ceilConstant const):
(JSC::B3::ConstFloatValue::floorConstant const):
* Source/JavaScriptCore/b3/B3Value.cpp:
(JSC::B3::Value::isRounded const):
* Source/JavaScriptCore/b3/testb3_3.cpp:
(testCeilArg):
(testCeilImm):
(testCeilMem):
(testCeilCeilArg):
(testFloorCeilArg):
(testCeilArgWithUselessDoubleConversion):
(testCeilArgWithEffectfulDoubleConversion):
(testFloorArg):
(testFloorImm):
(testFloorMem):
(testFloorFloorArg):
(testCeilFloorArg):
(testFloorArgWithUselessDoubleConversion):
(testFloorArgWithEffectfulDoubleConversion):
* Source/JavaScriptCore/inspector/scripts/tests/expected/command-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-async-attribute.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-optional-call-return-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/definitions-with-mac-platform.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-debuggableTypes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetTypes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domains-with-varying-command-sizes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/enum-values.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/event-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/events-with-optional-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/generate-domains-with-feature-guards.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/same-type-id-different-domain.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/shadowed-optional-type-setters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/should-strip-comments.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-aliased-primitive-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-array-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-enum-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-requiring-runtime-casts.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-with-open-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/version.json-result:
* Source/WTF/wtf/DateMath.h:
(WTF::jsCurrentTime):
(WTF::daysFrom1970ToYear):
(WTF::msToDays):
(WTF::dateToDaysFrom1970):
(WTF::msToMinutes):
(WTF::msToHours):
(WTF::msToSeconds):
* Source/WebCore/Modules/webaudio/AudioBufferSourceNode.cpp:
(WebCore::AudioBufferSourceNode::renderFromBuffer):
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::processARate):
(WebCore::OscillatorNode::processKRate):
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
(WebCore::isValueOutOfRangeForProperty):
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::constrainedAttachedWindowHeight):
(WebCore::InspectorFrontendClientLocal::constrainedAttachedWindowWidth):
* Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
(WebCore::Layout::InlineLevelBox::preferredLineHeight const):
(WebCore::Layout::InlineLevelBox::AscentAndDescent::round):
* Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp:
(WebCore::Layout::LineBoxBuilder::enclosingAscentDescentWithFallbackFonts const):
(WebCore::Layout::LineBoxBuilder::setLayoutBoundsForInlineBox const):
(WebCore::Layout::LineBoxBuilder::adjustInlineBoxHeightsForLineBoxContainIfApplicable):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendTextDisplayBox):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::breakWord):
* Source/WebCore/page/FrameSnapshotting.cpp:
(WebCore::snapshotFrameRectWithClip):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::resizePageRectsKeepingRatio):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::forceLayoutForPagination):
(WebCore::LocalFrameView::adjustPageHeightDeprecated):
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::computePageRectsWithPageSizeInternal):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::distanceDataForNode):
* Source/WebCore/page/ViewportConfiguration.cpp:
(WebCore::ViewportConfiguration::updateMinimumLayoutSize):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::renderRectForPoint const):
* Source/WebCore/page/scrolling/ThreadedScrollingTreeScrollingNodeDelegate.cpp:
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::adjustedScrollPosition const):
* Source/WebCore/platform/DateComponents.cpp:
(WebCore::DateComponents::setMillisecondsSinceMidnightInternal):
(WebCore::DateComponents::setMillisecondsSinceEpochForDate):
(WebCore::DateComponents::setMillisecondsSinceEpochForDateTimeLocal):
(WebCore::DateComponents::setMillisecondsSinceEpochForMonth):
(WebCore::DateComponents::setMillisecondsSinceMidnight):
(WebCore::DateComponents::setMonthsSinceEpoch):
(WebCore::DateComponents::setMillisecondsSinceEpochForWeek):
* Source/WebCore/platform/LayoutUnit.h:
(WebCore::LayoutUnit::fromFloatCeil):
(WebCore::LayoutUnit::fromFloatFloor):
(WebCore::roundToDevicePixel):
(WebCore::floorToDevicePixel):
(WebCore::ceilToDevicePixel):
* Source/WebCore/platform/ScrollAnimationKinetic.cpp:
(WebCore::ScrollAnimationKinetic::PerAxisData::animateScroll):
* Source/WebCore/platform/ScrollbarThemeComposite.cpp:
(WebCore::ScrollbarThemeComposite::thumbLength):
* Source/WebCore/platform/cocoa/DragImageCocoa.mm:
(WebCore::scaleDragImage):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::flooredIntSize):
(WebCore::roundedIntPoint):
(WebCore::flooredIntPoint):
(WebCore::ceiledIntPoint):
(WebCore::floorPointToDevicePixels):
(WebCore::ceilPointToDevicePixels):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::roundedIntSize):
(WebCore::flooredIntSize):
(WebCore::expandedIntSize):
(WebCore::flooredIntPoint):
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::fastAverageCharWidthIfAvailable const):
(WebCore::FontCascade::floatWidthForSimpleText const):
(WebCore::FontCascade::floatWidthForComplexText const):
* Source/WebCore/platform/graphics/FontCascade.h:
(WebCore::GlyphOverflow::extendTop):
(WebCore::GlyphOverflow::extendBottom):
* Source/WebCore/platform/graphics/GeometryUtilities.cpp:
(WebCore::normalizeAngleInRadians):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::computeLineBoundsAndAntialiasingModeForText):
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawTiled):
* Source/WebCore/platform/graphics/ImageBuffer.cpp:
(WebCore::ImageBuffer::sizeNeedsClamping):
* Source/WebCore/platform/graphics/ImageBufferBackend.cpp:
(WebCore::ImageBufferBackend::calculateBackendSize):
* Source/WebCore/platform/graphics/ShadowBlur.cpp:
(WebCore::calculateLobes):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::getStartDate const):
* Source/WebCore/platform/graphics/ca/TileController.cpp:
(WebCore::TileController::computeTileSize):
* Source/WebCore/platform/graphics/ca/TileGrid.cpp:
(WebCore::TileGrid::getTileIndexRangeForRect const):
* Source/WebCore/platform/graphics/cairo/CairoOperations.cpp:
(WebCore::Cairo::computeLineBoundsAndAntialiasingModeForText):
(WebCore::Cairo::drawSurface):
* Source/WebCore/platform/graphics/cairo/ImageBufferCairoBackend.cpp:
(WebCore::ImageBufferCairoBackend::transformToColorSpace):
* Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp:
(WebCore::GraphicsContextCG::strokePath):
(WebCore::GraphicsContextCG::strokeRect):
(WebCore::GraphicsContextCG::roundToDevicePixels const):
* Source/WebCore/platform/graphics/cg/IntRectCG.cpp:
(WebCore::enclosingIntRect):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::Font::platformInit):
* Source/WebCore/platform/graphics/filters/FEGaussianBlur.cpp:
(WebCore::clampedToKernelSize):
* Source/WebCore/platform/graphics/filters/software/FEColorMatrixSoftwareApplier.cpp:
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformAccelerated const):
* Source/WebCore/platform/graphics/filters/software/FETurbulenceSoftwareApplier.cpp:
(WebCore::FETurbulenceSoftwareApplier::computeStitching):
* Source/WebCore/platform/graphics/harfbuzz/ComplexTextControllerHarfBuzz.cpp:
(WebCore::ComplexTextController::collectComplexTextRunsForCharacters):
* Source/WebCore/platform/graphics/mac/LegacyDisplayRefreshMonitorMac.cpp:
(WebCore::LegacyDisplayRefreshMonitorMac::nominalFramesPerSecondFromDisplayLink):
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::MenuListButtonMac::draw):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
(WebCore::TransformationMatrix::clampedBoundsOfProjectedQuad const):
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
(WebCore::ComplexTextController::collectComplexTextRunsForCharacters):
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/ios/wak/WAKScrollView.mm:
(-[WAKScrollView scrollWheel:]):
* Source/WebCore/platform/ios/wak/WKView.mm:
(_WKViewAutoresizeCoord):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::setDragImageImpl):
* Source/WebCore/platform/mac/ScrollAnimationRubberBand.mm:
(WebCore::roundTowardZero):
(WebCore::roundToDevicePixelTowardZero):
* Source/WebCore/platform/mac/ScrollingEffectsController.mm:
(WebCore::ScrollingEffectsController::applyScrollDeltaWithStretching):
* Source/WebCore/rendering/AttachmentLayout.mm:
(WebCore::AttachmentLayout::layOutTitle):
(WebCore::AttachmentLayout::AttachmentLayout):
* Source/WebCore/rendering/BorderEdge.cpp:
(WebCore::BorderEdge::BorderEdge):
* Source/WebCore/rendering/CaretRectComputation.cpp:
(WebCore::computeCaretRectForLinePosition):
* Source/WebCore/rendering/LegacyInlineFlowBox.cpp:
(WebCore::LegacyInlineFlowBox::computeLogicalBoxHeights):
* Source/WebCore/rendering/LegacyInlineFlowBox.h:
(WebCore::LegacyInlineFlowBox::logicalRightVisualOverflow const):
* Source/WebCore/rendering/LegacyRootInlineBox.cpp:
(WebCore::LegacyRootInlineBox::lineSnapAdjustment const):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::adjustLogicalRightOffsetForLine const):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::adjustComputedFontSizes):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::paintReplaced):
(WebCore::RenderEmbeddedObject::getReplacementTextGeometry const):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::computeIntrinsicLogicalWidths const):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::setImageSizeForAltText):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::updateOptionsWidth):
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::columnCount const):
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::scaleEmToUnits const):
* Source/WebCore/rendering/RenderTextControlMultiLine.cpp:
(WebCore::RenderTextControlMultiLine::preferredContentLogicalWidth const):
* Source/WebCore/rendering/RenderTextControlSingleLine.cpp:
(WebCore::RenderTextControlSingleLine::preferredContentLogicalWidth const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintMenuListButtonDecorations):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::write):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::viewHeight const):
(WebCore::RenderView::viewWidth const):
* Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp:
(WebCore::RenderMathMLFraction::firstLineBaseline const):
* Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp:
(WebCore::RenderMathMLScripts::firstLineBaseline const):
* Source/WebCore/rendering/style/ShadowData.h:
(WebCore::ShadowData::paintingExtent const):
* Source/WebCore/rendering/svg/LegacyRenderSVGForeignObject.cpp:
(WebCore::LegacyRenderSVGForeignObject::updateLogicalWidth):
(WebCore::LegacyRenderSVGForeignObject::computeLogicalHeight const):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::currentValuesForValuesAnimation):
* Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp:
(WebCore::SVGFEConvolveMatrixElement::isValidTargetXOffset const):
(WebCore::SVGFEConvolveMatrixElement::isValidTargetYOffset const):
(WebCore::SVGFEConvolveMatrixElement::createFilterEffect const):
* Source/WebCore/svg/SVGFontFaceElement.cpp:
(WebCore::SVGFontFaceElement::unitsPerEm const):
(WebCore::SVGFontFaceElement::xHeight const):
(WebCore::SVGFontFaceElement::capHeight const):
(WebCore::SVGFontFaceElement::ascent const):
(WebCore::SVGFontFaceElement::descent const):
* Source/WebCore/svg/SVGPathParser.cpp:
(WebCore::SVGPathParser::decomposeArcToCubic):
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::calculateAnimationPercentAndRepeat const):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.h:
* Source/WebCore/xml/XPathFunctions.cpp:
(WebCore::XPath::FunFloor::evaluate const):
(WebCore::XPath::FunCeiling::evaluate const):
* Source/WebKit/Shared/CoordinatedGraphics/SimpleViewportController.cpp:
(WebKit::SimpleViewportController::boundContentsPositionAtScale const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.mm:
(WebKit::GraphicsContextCGDisplayList::GraphicsContextCGDisplayList):
* Source/WebKit/UIProcess/API/glib/IconDatabase.cpp:
(WebKit::IconDatabase::populatePageURLToIconURLMap):
(WebKit::IconDatabase::pruneTimerFired):
(WebKit::IconDatabase::iconIDForIconURL):
* Source/WebKit/UIProcess/_WKTouchEventGenerator.mm:
(-[_WKTouchEventGenerator _createIOHIDEventType:]):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
(WebKit::ViewGestureController::snapshot):
(WebKit::ViewGestureController::draw):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):
* Source/WebKit/UIProcess/mac/DisplayLink.cpp:
(WebKit::DisplayLink::nominalFramesPerSecondFromDisplayLink):
* Source/WebKit/UIProcess/mac/ViewGestureControllerMac.mm:
(WebKit::ViewGestureController::handleSwipeGesture):
* Source/WebKit/UIProcess/mac/WebPopupMenuProxyMac.mm:
(WebKit::WebPopupMenuProxyMac::showPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/win/WebPopupMenuWin.cpp:
(WebKit::WebPopupMenu::setUpPlatformData):
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
(WebKit::MomentumEventDispatcher::offsetAtTime):
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
(WebKit::ViewGestureGeometryCollector::computeTextLegibilityScales):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::computePagesForPrintingPDFDocument):
(WebKit::drawPDFPage):
(WebKit::WebPage::drawPDFDocument):
(WebKit::roundCGFloat): Deleted.
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
(-[WebPDFView _computePageRects]):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder _getPDFPageBounds:]):
(-[WebPDFViewPlaceholder _computePageRects:]):
* Source/WebKitLegacy/mac/WebCoreSupport/PopupMenuMac.mm:
(PopupMenuMac::show):
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
(-[WebDynamicScrollBarsView scrollClipView:toPoint:]):
(-[WebDynamicScrollBarsView updateScrollers]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView knowsPageRange:]):
(imageFromRect):
* Source/WebKitLegacy/mac/WebView/WebTextCompletionController.mm:
(-[WebTextCompletionController _placePopupWindow:]):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(-[ScrollViewResizerDelegate view:didSetFrame:oldFrame:asResultOfZoom:]):
* Tools/TestWebKitAPI/Tests/ios/TextAutosizingBoost.mm:
(TEST):
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
(-[HIDEventGenerator _createIOHIDEventType:]):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0eb313d248713c1a73f43a85ead7ad0e3265f55c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9970 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10216 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10468 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11621 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9655 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9979 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12204 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10168 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12599 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10125 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10910 "1 flakes 3 failures") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8440 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12005 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8217 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9034 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16371 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/8444 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9316 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9183 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12455 "Passed tests") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/9466 "JSC test binary failure: testapi") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9687 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7864 "5 failures") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10109 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8846 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2594 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13075 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/10381 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9465 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2575 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -40,7 +40,8 @@ class LineBoxVerticalAligner;
 class InlineLevelBox {
 public:
-    enum class LineSpanningInlineBox { Yes, No };
+    enum class LineSpanningInlineBox { Yes,
```
Please revert this change.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -147,7 +148,7 @@ class InlineLevelBox {
     // This is the combination of margin and border boxes. Inline level boxes are vertically aligned using their margin boxes.
     InlineRect m_logicalRect;
-    std::optional<LayoutBounds> m_layoutBounds { };
+    std::optional<LayoutBounds> m_layoutBounds {};
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -165,25 +166,26 @@ class InlineLevelBox {
         WTF::OptionSet<LineBoxContain> lineBoxContain;
         InlineLayoutUnit primaryFontSize { 0 };
-        VerticalAlignment verticalAlignment { };
+        VerticalAlignment verticalAlignment {};
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -165,25 +166,26 @@ class InlineLevelBox {
 
     struct Annotation {
-        enum class Type : uint8_t { Above, Under };
+        enum class Type : uint8_t { Above,
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -165,25 +166,26 @@ class InlineLevelBox {
+        enum class Type : uint8_t { Above,
+            Under };
         Type type { Type::Above };
-        InlineLayoutUnit size { };
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -165,25 +166,26 @@ class InlineLevelBox {
 inline InlineLevelBox::InlineLevelBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft, InlineLayoutSize logicalSize, Type type, OptionSet<PositionWithinLayoutBox> positionWithinLayoutBox)
     : m_layoutBox(layoutBox)
-    , m_logicalRect({ }, logicalLeft, logicalSize.width(), logicalSize.height())
+    , m_logicalRect({}, logicalLeft, logicalSize.width(), logicalSize.height())
```
All these changes need to be reverted.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -242,27 +243,27 @@ inline bool InlineLevelBox::hasLineBoxRelativeAlignment() const
 inline InlineLevelBox InlineLevelBox::createAtomicInlineLevelBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft, InlineLayoutUnit logicalWidth)
 {
-    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, { } }, Type::AtomicInlineLevelBox };
+    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, {} }, Type::AtomicInlineLevelBox };
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -242,27 +243,27 @@ inline bool InlineLevelBox::hasLineBoxRelativeAlignment() const
 inline InlineLevelBox InlineLevelBox::createRootInlineBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft, InlineLayoutUnit logicalWidth)
 {
-    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, { } }, Type::RootInlineBox, { } };
+    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, {} }, Type::RootInlineBox, {} };
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -242,27 +243,27 @@ inline bool InlineLevelBox::hasLineBoxRelativeAlignment() const
 inline InlineLevelBox InlineLevelBox::createInlineBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft, InlineLayoutUnit logicalWidth, LineSpanningInlineBox isLineSpanning)
 {
-    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, { } }, isLineSpanning == LineSpanningInlineBox::Yes ? Type::LineSpanningInlineBox : Type::InlineBox, { } };
+    return InlineLevelBox { layoutBox, style, logicalLeft, { logicalWidth, {} }, isLineSpanning == LineSpanningInlineBox::Yes ? Type::LineSpanningInlineBox : Type::InlineBox, {} };
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -242,27 +243,27 @@ inline bool InlineLevelBox::hasLineBoxRelativeAlignment() const
 inline InlineLevelBox InlineLevelBox::createLineBreakBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft)
 {
-    return InlineLevelBox { layoutBox, style, logicalLeft, { }, Type::LineBreakBox };
+    return InlineLevelBox { layoutBox, style, logicalLeft, {}, Type::LineBreakBox };
```
Ditto.

rniwa commented on Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
```diff
@@ -242,27 +243,27 @@ inline bool InlineLevelBox::hasLineBoxRelativeAlignment() const
 inline InlineLevelBox InlineLevelBox::createGenericInlineLevelBox(const Box& layoutBox, const RenderStyle& style, InlineLayoutUnit logicalLeft)
 {
-    return InlineLevelBox { layoutBox, style, logicalLeft, { }, Type::GenericInlineLevelBox };
+    return InlineLevelBox { layoutBox, style, logicalLeft, {}, Type::GenericInlineLevelBox };
```
Ditto.

rniwa commented on Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:
```diff
@@ -124,18 +124,6 @@ typedef struct CGSRegionObject* CGSRegionObj;
 typedef struct CGSRegionObject* CGRegionRef;
 #endif
 
-#ifdef CGFLOAT_IS_DOUBLE
```
I don't think we should remove these, nor should we replace the existing use of them.

AZero13 replied:
Resolved!

LumioseSil committed:
Prefer std::ceil, std::floor, and std::round over their C counterparts
https://bugs.webkit.org/show_bug.cgi?id=253963

Reviewed by NOBODY (OOPS!).

Unlike the C library versions, these std functions have overloads for
every type, so one does not need to worry about which version of the
function to call.

Continuation of https://commits.webkit.org/261584@main

* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
(JSC::B3::ConstFloatValue::ceilConstant const):
(JSC::B3::ConstFloatValue::floorConstant const):
* Source/JavaScriptCore/b3/B3Value.cpp:
(JSC::B3::Value::isRounded const):
* Source/JavaScriptCore/b3/testb3_3.cpp:
(testCeilArg):
(testCeilImm):
(testCeilMem):
(testCeilCeilArg):
(testFloorCeilArg):
(testCeilArgWithUselessDoubleConversion):
(testCeilArgWithEffectfulDoubleConversion):
(testFloorArg):
(testFloorImm):
(testFloorMem):
(testFloorFloorArg):
(testCeilFloorArg):
(testFloorArgWithUselessDoubleConversion):
(testFloorArgWithEffectfulDoubleConversion):
* Source/JavaScriptCore/inspector/scripts/tests/expected/command-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-async-attribute.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/commands-with-optional-call-return-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/definitions-with-mac-platform.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-debuggableTypes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetTypes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domains-with-varying-command-sizes.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/enum-values.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/event-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/events-with-optional-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/generate-domains-with-feature-guards.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/same-type-id-different-domain.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/shadowed-optional-type-setters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/should-strip-comments.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-aliased-primitive-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-array-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-enum-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-declaration-object-type.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-requiring-runtime-casts.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/type-with-open-parameters.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/version.json-result:
* Source/WTF/wtf/DateMath.h:
(WTF::jsCurrentTime):
(WTF::daysFrom1970ToYear):
(WTF::msToDays):
(WTF::dateToDaysFrom1970):
(WTF::msToMinutes):
(WTF::msToHours):
(WTF::msToSeconds):
* Source/WebCore/Modules/webaudio/AudioBufferSourceNode.cpp:
(WebCore::AudioBufferSourceNode::renderFromBuffer):
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::processARate):
(WebCore::OscillatorNode::processKRate):
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
(WebCore::isValueOutOfRangeForProperty):
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::constrainedAttachedWindowHeight):
(WebCore::InspectorFrontendClientLocal::constrainedAttachedWindowWidth):
* Source/WebCore/layout/formattingContexts/inline/InlineLevelBox.h:
(WebCore::Layout::InlineLevelBox::preferredLineHeight const):
(WebCore::Layout::InlineLevelBox::AscentAndDescent::round):
* Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp:
(WebCore::Layout::LineBoxBuilder::enclosingAscentDescentWithFallbackFonts const):
(WebCore::Layout::LineBoxBuilder::setLayoutBoundsForInlineBox const):
(WebCore::Layout::LineBoxBuilder::adjustInlineBoxHeightsForLineBoxContainIfApplicable):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendTextDisplayBox):
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(WebCore::Layout::TextUtil::breakWord):
* Source/WebCore/page/FrameSnapshotting.cpp:
(WebCore::snapshotFrameRectWithClip):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::resizePageRectsKeepingRatio):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::forceLayoutForPagination):
(WebCore::LocalFrameView::adjustPageHeightDeprecated):
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::computePageRectsWithPageSizeInternal):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::distanceDataForNode):
* Source/WebCore/page/ViewportConfiguration.cpp:
(WebCore::ViewportConfiguration::updateMinimumLayoutSize):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::renderRectForPoint const):
* Source/WebCore/page/scrolling/ThreadedScrollingTreeScrollingNodeDelegate.cpp:
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::adjustedScrollPosition const):
* Source/WebCore/platform/DateComponents.cpp:
(WebCore::DateComponents::setMillisecondsSinceMidnightInternal):
(WebCore::DateComponents::setMillisecondsSinceEpochForDate):
(WebCore::DateComponents::setMillisecondsSinceEpochForDateTimeLocal):
(WebCore::DateComponents::setMillisecondsSinceEpochForMonth):
(WebCore::DateComponents::setMillisecondsSinceMidnight):
(WebCore::DateComponents::setMonthsSinceEpoch):
(WebCore::DateComponents::setMillisecondsSinceEpochForWeek):
* Source/WebCore/platform/LayoutUnit.h:
(WebCore::LayoutUnit::fromFloatCeil):
(WebCore::LayoutUnit::fromFloatFloor):
(WebCore::roundToDevicePixel):
(WebCore::floorToDevicePixel):
(WebCore::ceilToDevicePixel):
* Source/WebCore/platform/ScrollAnimationKinetic.cpp:
(WebCore::ScrollAnimationKinetic::PerAxisData::animateScroll):
* Source/WebCore/platform/ScrollbarThemeComposite.cpp:
(WebCore::ScrollbarThemeComposite::thumbLength):
* Source/WebCore/platform/cocoa/DragImageCocoa.mm:
(WebCore::scaleDragImage):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::flooredIntSize):
(WebCore::roundedIntPoint):
(WebCore::flooredIntPoint):
(WebCore::ceiledIntPoint):
(WebCore::floorPointToDevicePixels):
(WebCore::ceilPointToDevicePixels):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::roundedIntSize):
(WebCore::flooredIntSize):
(WebCore::expandedIntSize):
(WebCore::flooredIntPoint):
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::fastAverageCharWidthIfAvailable const):
(WebCore::FontCascade::floatWidthForSimpleText const):
(WebCore::FontCascade::floatWidthForComplexText const):
* Source/WebCore/platform/graphics/FontCascade.h:
(WebCore::GlyphOverflow::extendTop):
(WebCore::GlyphOverflow::extendBottom):
* Source/WebCore/platform/graphics/GeometryUtilities.cpp:
(WebCore::normalizeAngleInRadians):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::computeLineBoundsAndAntialiasingModeForText):
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawTiled):
* Source/WebCore/platform/graphics/ImageBuffer.cpp:
(WebCore::ImageBuffer::sizeNeedsClamping):
* Source/WebCore/platform/graphics/ImageBufferBackend.cpp:
(WebCore::ImageBufferBackend::calculateBackendSize):
* Source/WebCore/platform/graphics/ShadowBlur.cpp:
(WebCore::calculateLobes):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::getStartDate const):
* Source/WebCore/platform/graphics/ca/TileController.cpp:
(WebCore::TileController::computeTileSize):
* Source/WebCore/platform/graphics/ca/TileGrid.cpp:
(WebCore::TileGrid::getTileIndexRangeForRect const):
* Source/WebCore/platform/graphics/cairo/CairoOperations.cpp:
(WebCore::Cairo::computeLineBoundsAndAntialiasingModeForText):
(WebCore::Cairo::drawSurface):
* Source/WebCore/platform/graphics/cairo/ImageBufferCairoBackend.cpp:
(WebCore::ImageBufferCairoBackend::transformToColorSpace):
* Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp:
(WebCore::GraphicsContextCG::strokePath):
(WebCore::GraphicsContextCG::strokeRect):
(WebCore::GraphicsContextCG::roundToDevicePixels const):
* Source/WebCore/platform/graphics/cg/IntRectCG.cpp:
(WebCore::enclosingIntRect):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::Font::platformInit):
* Source/WebCore/platform/graphics/filters/FEGaussianBlur.cpp:
(WebCore::clampedToKernelSize):
* Source/WebCore/platform/graphics/filters/software/FEColorMatrixSoftwareApplier.cpp:
(WebCore::FEColorMatrixSoftwareApplier::applyPlatformAccelerated const):
* Source/WebCore/platform/graphics/filters/software/FETurbulenceSoftwareApplier.cpp:
(WebCore::FETurbulenceSoftwareApplier::computeStitching):
* Source/WebCore/platform/graphics/harfbuzz/ComplexTextControllerHarfBuzz.cpp:
(WebCore::ComplexTextController::collectComplexTextRunsForCharacters):
* Source/WebCore/platform/graphics/mac/LegacyDisplayRefreshMonitorMac.cpp:
(WebCore::LegacyDisplayRefreshMonitorMac::nominalFramesPerSecondFromDisplayLink):
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::MenuListButtonMac::draw):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
(WebCore::TransformationMatrix::clampedBoundsOfProjectedQuad const):
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
(WebCore::ComplexTextController::collectComplexTextRunsForCharacters):
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/ios/wak/WAKScrollView.mm:
(-[WAKScrollView scrollWheel:]):
* Source/WebCore/platform/ios/wak/WKView.mm:
(_WKViewAutoresizeCoord):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::setDragImageImpl):
* Source/WebCore/platform/mac/ScrollAnimationRubberBand.mm:
(WebCore::roundTowardZero):
(WebCore::roundToDevicePixelTowardZero):
* Source/WebCore/platform/mac/ScrollingEffectsController.mm:
(WebCore::ScrollingEffectsController::applyScrollDeltaWithStretching):
* Source/WebCore/rendering/AttachmentLayout.mm:
(WebCore::AttachmentLayout::layOutTitle):
(WebCore::AttachmentLayout::AttachmentLayout):
* Source/WebCore/rendering/BorderEdge.cpp:
(WebCore::BorderEdge::BorderEdge):
* Source/WebCore/rendering/CaretRectComputation.cpp:
(WebCore::computeCaretRectForLinePosition):
* Source/WebCore/rendering/LegacyInlineFlowBox.cpp:
(WebCore::LegacyInlineFlowBox::computeLogicalBoxHeights):
* Source/WebCore/rendering/LegacyInlineFlowBox.h:
(WebCore::LegacyInlineFlowBox::logicalRightVisualOverflow const):
* Source/WebCore/rendering/LegacyRootInlineBox.cpp:
(WebCore::LegacyRootInlineBox::lineSnapAdjustment const):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::adjustLogicalRightOffsetForLine const):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::adjustComputedFontSizes):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::paintReplaced):
(WebCore::RenderEmbeddedObject::getReplacementTextGeometry const):
* Source/WebCore/rendering/RenderFileUploadControl.cpp:
(WebCore::RenderFileUploadControl::computeIntrinsicLogicalWidths const):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::setImageSizeForAltText):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::updateOptionsWidth):
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::columnCount const):
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderTextControl.cpp:
(WebCore::RenderTextControl::scaleEmToUnits const):
* Source/WebCore/rendering/RenderTextControlMultiLine.cpp:
(WebCore::RenderTextControlMultiLine::preferredContentLogicalWidth const):
* Source/WebCore/rendering/RenderTextControlSingleLine.cpp:
(WebCore::RenderTextControlSingleLine::preferredContentLogicalWidth const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paintSliderTicks):
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintMenuListButtonDecorations):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::write):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::viewHeight const):
(WebCore::RenderView::viewWidth const):
* Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp:
(WebCore::RenderMathMLFraction::firstLineBaseline const):
* Source/WebCore/rendering/mathml/RenderMathMLScripts.cpp:
(WebCore::RenderMathMLScripts::firstLineBaseline const):
* Source/WebCore/rendering/style/ShadowData.h:
(WebCore::ShadowData::paintingExtent const):
* Source/WebCore/rendering/svg/LegacyRenderSVGForeignObject.cpp:
(WebCore::LegacyRenderSVGForeignObject::updateLogicalWidth):
(WebCore::LegacyRenderSVGForeignObject::computeLogicalHeight const):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::currentValuesForValuesAnimation):
* Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp:
(WebCore::SVGFEConvolveMatrixElement::isValidTargetXOffset const):
(WebCore::SVGFEConvolveMatrixElement::isValidTargetYOffset const):
(WebCore::SVGFEConvolveMatrixElement::createFilterEffect const):
* Source/WebCore/svg/SVGFontFaceElement.cpp:
(WebCore::SVGFontFaceElement::unitsPerEm const):
(WebCore::SVGFontFaceElement::xHeight const):
(WebCore::SVGFontFaceElement::capHeight const):
(WebCore::SVGFontFaceElement::ascent const):
(WebCore::SVGFontFaceElement::descent const):
* Source/WebCore/svg/SVGPathParser.cpp:
(WebCore::SVGPathParser::decomposeArcToCubic):
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::calculateAnimationPercentAndRepeat const):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.h:
* Source/WebCore/xml/XPathFunctions.cpp:
(WebCore::XPath::FunFloor::evaluate const):
(WebCore::XPath::FunCeiling::evaluate const):
* Source/WebKit/Shared/CoordinatedGraphics/SimpleViewportController.cpp:
(WebKit::SimpleViewportController::boundContentsPositionAtScale const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.mm:
(WebKit::GraphicsContextCGDisplayList::GraphicsContextCGDisplayList):
* Source/WebKit/UIProcess/API/glib/IconDatabase.cpp:
(WebKit::IconDatabase::populatePageURLToIconURLMap):
(WebKit::IconDatabase::pruneTimerFired):
(WebKit::IconDatabase::iconIDForIconURL):
* Source/WebKit/UIProcess/_WKTouchEventGenerator.mm:
(-[_WKTouchEventGenerator _createIOHIDEventType:]):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
(WebKit::ViewGestureController::snapshot):
(WebKit::ViewGestureController::draw):
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):
* Source/WebKit/UIProcess/mac/DisplayLink.cpp:
(WebKit::DisplayLink::nominalFramesPerSecondFromDisplayLink):
* Source/WebKit/UIProcess/mac/ViewGestureControllerMac.mm:
(WebKit::ViewGestureController::handleSwipeGesture):
* Source/WebKit/UIProcess/mac/WebPopupMenuProxyMac.mm:
(WebKit::WebPopupMenuProxyMac::showPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/win/WebPopupMenuWin.cpp:
(WebKit::WebPopupMenu::setUpPlatformData):
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
(WebKit::MomentumEventDispatcher::offsetAtTime):
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp:
(WebKit::ViewGestureGeometryCollector::computeTextLegibilityScales):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::computePagesForPrintingPDFDocument):
(WebKit::drawPDFPage):
(WebKit::WebPage::drawPDFDocument):
(WebKit::roundCGFloat): Deleted.
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
(-[WebPDFView _computePageRects]):
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
(-[WebPDFViewPlaceholder _getPDFPageBounds:]):
(-[WebPDFViewPlaceholder _computePageRects:]):
* Source/WebKitLegacy/mac/WebCoreSupport/PopupMenuMac.mm:
(PopupMenuMac::show):
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
(-[WebDynamicScrollBarsView scrollClipView:toPoint:]):
(-[WebDynamicScrollBarsView updateScrollers]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView knowsPageRange:]):
(imageFromRect):
* Source/WebKitLegacy/mac/WebView/WebTextCompletionController.mm:
(-[WebTextCompletionController _placePopupWindow:]):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(-[ScrollViewResizerDelegate view:didSetFrame:oldFrame:asResultOfZoom:]):
* Tools/TestWebKitAPI/Tests/ios/TextAutosizingBoost.mm:
(TEST):
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
(-[HIDEventGenerator _createIOHIDEventType:]):

rniwa commented on Source/WebCore/platform/graphics/cg/IntRectCG.cpp:
```diff
@@ -39,11 +39,11 @@ IntRect::operator CGRect() const
-    int r = static_cast<int>(ceilf(CGRectGetMaxX(rect)));
-    int b = static_cast<int>(ceilf(CGRectGetMaxY(rect)));
-    return IntRect(l, t, r - l, b - t);
+    int L = static_cast<int>(std::floor(rect.origin.x));
```
Why are we capitalizing these? I don't think we should.

AZero13 replied:
style complains otherwise but yeah I will change back.

JonWBedard commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
Seems like we're doing the exact opposite of the PR title here: we're taking something that is `std::abs` and using `CGFAbs` instead, that doesn't feel deliberate.

AZero13 replied:
That was for consistency but yeah.

rniwa commented on Source/WebCore/platform/ios/LegacyTileGrid.mm:
```diff
@@ -302,8 +302,8 @@ IntRect rect(index.x() * m_tileSize.width() - (m_origin.x() ? m_tileSize.width()
     CGRect tileHostBounds = [hostLayer convertRect:[hostLayer bounds] toLayer:tileHostLayer()];
     CGSize transformedSize;
-    transformedSize.width = CGRound(tileHostBounds.size.width);
-    transformedSize.height = CGRound(tileHostBounds.size.height);
```
I don't think we should replace this use of CGRound with std::round.

JonWBedard commented on Source/WebCore/xml/XPathFunctions.cpp:
This file now contains different changes that what were reviewed and which are out of scope of the change title.

JonWBedard commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
```diff
@@ -443,8 +443,8 @@ static inline CGFloat floorToDevicePixel(CGFloat input, float deviceScaleFactor)
 {
-    return std::abs(a.x * deviceScaleFactor - b.x * deviceScaleFactor) < std::numeric_limits<float>::epsilon()
-        && std::abs(a.y * deviceScaleFactor - b.y * deviceScaleFactor) < std::numeric_limits<float>::epsilon();
+    return CGFAbs(a.x * deviceScaleFactor - b.x * deviceScaleFactor) < std::numeric_limits<float>::epsilon()
```
This is doing the opposite of the change title, it replaces `std::abs` with `CGFAbs`.

rniwa commented on Source/WebCore/rendering/RenderThemeIOS.mm:
```diff
@@ -1584,7 +1584,7 @@ static bool shouldUseConvexGradient(const Color& backgroundColor)
         CGFloat matchRatio = (attachmentIconSize / iconSize) - 1.0f;
         if (matchRatio < 0.3f) {
-            matchRatio = CGFAbs(matchRatio);
+            matchRatio = std::abs(matchRatio);
```
I don't think we should replace an existing use of `CGFAbs` with `std::abs`.

rniwa replied:
This comment is still not addressed.

rniwa commented on Source/WebCore/xml/XPathFunctions.cpp:
```diff
@@ -706,7 +706,8 @@ double FunRound::round(double val)
 Value FunRound::evaluate() const
 {
-    return round(argument(0).evaluate().toNumber());
+    // Why does std::round not work here?
```
I don't think we should add this comment either.

rniwa commented on Source/WebCore/xml/XPathFunctions.cpp:
```diff
@@ -706,7 +706,8 @@ double FunRound::round(double val)
 {
-    return round(argument(0).evaluate().toNumber());
+    // Why does std::round not work here?
+    return FunRound::round(argument(0).evaluate().toNumber());
```
No need to qualify the function in the same class so we should just keep the original code.

AZero13 replied:
I got this confused with std::round so yeah we should keep this.

rniwa commented on Source/WebKit/UIProcess/API/Cocoa/_WKThumbnailView.mm:
```diff
@@ -146,7 +146,7 @@ - (void)requestSnapshot
         if (_maximumSnapshotSize.height)
             sizeConstraintScale = CGFloatMin(sizeConstraintScale, _maximumSnapshotSize.height / bitmapSize.height());
-        bitmapSize = WebCore::IntSize(CGCeiling(bitmapSize.width() * sizeConstraintScale), CGCeiling(bitmapSize.height() * sizeConstraintScale));
+        bitmapSize = WebCore::IntSize(std::ceil(bitmapSize.width() * sizeConstraintScale), std::ceil(bitmapSize.height() * sizeConstraintScale));
```
Again, I don't think we should replace uses of CGCeiling with std::ceil.

JonWBedard commented on Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
```diff
@@ -97,6 +97,7 @@
 #import <WebCore/ThemeMac.h>
 #import <WebCore/VisibleUnits.h>
 #import <WebCore/WindowsKeyboardCodes.h>
+#import <pal/spi/cg/CoreGraphicsSPI.h>
```
Why are we adding this import? It's unrelated to the stated purpose of the change.

JonWBedard commented on Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp:
```diff
@@ -216,12 +216,13 @@ void SVGFEConvolveMatrixElement::svgAttributeChanged(const QualifiedName& attrNa
+    int orderYValue = 3;
+    if (hasAttribute(SVGNames::orderAttr)) {
+        orderXValue = orderX();
+        orderYValue = orderY();
```
This change has nothing to do with the stated PR and should not be included.

JonWBedard commented on Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp:
```diff
@@ -241,19 +242,19 @@ RefPtr<FilterEffect> SVGFEConvolveMatrixElement::createFilterEffect(const Filter
+    int kernelUnitLengthYValue = 1;
+    if (hasAttribute(SVGNames::kernelUnitLengthAttr)) {
+        kernelUnitLengthXValue = kernelUnitLengthX();
+        kernelUnitLengthYValue = kernelUnitLengthY();
```
This change has nothing to do with the stated PR and should not be included.

JonWBedard reviewed (commented):
This is already quite a large PR. The most recent update changed 30+ files, none of which (as far as I can tell) have anything to do with std::ceil, std::floor and std::round.
